src/aind_slims_api/core.py

```diff
@@ -199,6 +204,44 @@ def fetch(
 
         return records
 
+    def fetch_models(
+        self,
+        model: Type[SlimsBaseModelTypeVar],
+        *args,
+        sort: Optional[str | list[str]] = None,
+        start: Optional[int] = None,
+        end: Optional[int] = None,
+        **kwargs,
+    ) -> tuple[list[SlimsBaseModelTypeVar], list[dict[str, Any]]]:
+        """Fetch records from SLIMS and return them as SlimsBaseModel objects
+
+        Returns
+        -------
+        tuple:
+            list:
+                Validated SlimsBaseModel objects
+            list:
+                Dictionaries representations of objects that failed validation
+        """
+        response = self.fetch(
+            model._slims_table.default,  # TODO: consider changing fetch method
+            *args,
+            sort=sort,
+            start=start,
+            end=end,
+            **kwargs,
+        )
+        validated = []
+        unvalidated = []
+        for record in response:
+            try:
+                validated.append(model.model_validate(record))
+            except ValidationError as e:
+                logger.error(f"SLIMS data validation failed, {repr(e)}")
+                unvalidated.append(record.json_entity)
```

Comment:
Now that I think about it a little more, I don't think this method should return two lists. Doing that kind of splits the difference between using models and using dicts, and I think if we want to use the models we should fully commit. 

Might be useful to think through the use cases of when model validation might fail, because ideally the database has proper data (especially if it's been written from the aind_slims_api). 
1. Data got fed to the database manually through ui or through some other process (not this library), and our models are more restricted than the SLIMS fields. 
2. The models got updated and old data doesn't match
3. The database schema got updated and the models don't match.

I think these are all things that we'd want to catch with integration test errors, but shouldn't expect users to have to deal with.

Anyway, maybe more discussion is warranted, but I'm now thinking validation errors should raise exceptions (maybe we'd want to raise one error for all the failures like Pydantic does; maybe TypeAdapter is appropriate?)